### ISSUE TITLE
Attribute 'processPath' is required. IIS InProcess SCD hosting

### ISSUE THREAD

**Issue by jpseini** [aspnet/AspNetCore#11738]
### Describe the bug
When trying to host an aspnet core web api in ISS (tried both 2.2 and 3.0), I am following the steps as per here: [Host in IIS](https://docs.microsoft.com/en-us/aspnet/core/host-and-deploy/iis/?view=aspnetcore-3.0#create-the-iis-site). When I try browse to the URL, I get a HTTP 500 error, and in my EventViewer I see the following message:
_Could not load configuration. Exception message:
Attribute 'processPath' is required._
With a source of _IIS AspNetCore Module V2_

### To Reproduce
1. Using version 2.2 or 3
2. Create a new ASP.Net Core Web application, using the API project template (I disabled HTTPS)
3. Edit the csproj file and make sure the HostingModel is set to InProcess, or add the following node under <PropertyGroup> if not already present: <AspNetCoreHostingModel>InProcess</AspNetCoreHostingModel>
4. I added UseIIS() to the CreateWebHostBuilder method
` public static IWebHostBuilder CreateWebHostBuilder(string[] args) =>
            WebHost.CreateDefaultBuilder(args)
                .UseIIS()
                .UseStartup<Startup>();`
5. Publish the project to a Folder. Under Advanced, set DeploymentMode to "Self Contained" and the Target Runtime to win-x86.
6. On the local machine (dev machine), .Net Core Hosting bundle 2.2 and 3.0 has been installed.
7. I created a folder in c:\inetpub\wwwroot (called test22) and copied the output of the publish from step 4 into this folder.
8. Follow the steps mentioned in above link. Create website, with physical path pointing to c:\inetpub\wwwroot\test22. Updated application pool to have "No Managed Code", and also updated the app pool identity to be my AD user (who is local admin), just in case it was permission related. Also set "Enable 32-bit Applications" to **true**.
9. Browse to http://localhost/test22/api/values, from the same dev machine, and get an HTTP 500 error
10. Check event viewer, see the error mentioned above.

### Expected behavior
The API should be called and values should be returned

### Screenshots
The web.config in the site folder (c:\inetpub\wwwroot\test22), looks as follows:
```
<configuration>
  <location path="." inheritInChildApplications="false">
    <system.webServer>
      <handlers>
        <add name="aspNetCore" path="*" verb="*" modules="AspNetCoreModuleV2" resourceType="Unspecified" />
      </handlers>
      <aspNetCore processPath=".\IISTest22.exe" stdoutLogEnabled="false" stdoutLogFile=".\logs\stdout" hostingModel="InProcess" />
    </system.webServer>
  </location>
</configuration>
```

### Additional context
I'm running on windows 10, IIS version 10.0.17763.1. 
I've tried a variety of variations on the above steps, all resulting in the same error. I've tried:
- Setting permission to full control on the folder
- Tried using ApplicationPoolIdentity as the pool identity instead of my AD account
- Having the physical path not in c:\inetput\wwwroot
- Performing a Web Publish, instead of publishing to a folder and then copying.

I can literally change the processPath attribute in the web config to anything and I get the same error, so it appears as though IIS is not able to find the config, or or parse it correctly.

I'm not sure if this is related to a configuration on my machine (although I have followed the steps in the docs), or if its a bug somewhere. I've been unable to find anyone else online who has experienced the same issue, or any other information related to the error message.

**Comment by jkotalik** [comment]
Hm, I've never seen this issue before. The error you are getting occurs when ANCM tries to get the processPath value and is empty. So your hypothesis about IIS not finding the config is correct.

My gut says there may be another declaration of the aspNetCore element, potentially somewhere in the global applicationhost.config in `C:\Windows\System32\inetsrv\config`. Look inside of that file for aspNetCore. There should be at least one place where the aspNetCore section is defined, but there could be a second one which is causing this issue.

**Comment by jpseini** [comment]
Thanks for the feedback @jkotalik. Unfortunately I don't see any duplicates. I searched the file for "aspnetcore" and only found the following results (only showing relevent nodes below, but obviously the file contains a lot of other info):

```
<sectionGroup name="system.webServer">
  <section name="aspNetCore" overrideModeDefault="Allow" />
</sectionGroup>
```

```
<globalModules>
   <add name="AspNetCoreModule" image="%SystemRoot%\system32\inetsrv\aspnetcore.dll" />
   <add name="AspNetCoreModuleV2" image="%ProgramFiles%\IIS\Asp.Net Core Module\V2\aspnetcorev2.dll" />
</globalModules>
```

```
<modules>
   <add name="AspNetCoreModule" />
   <add name="AspNetCoreModuleV2" />
</modules>
```

Does that look right? Any other ideas? Otherwise, my next step is to investigate and see if there is any logging I can turn on which will give me any idea where it is trying to find the web.config.

Thanks again.

**Comment by jkotalik** [comment]
The only other thing I can think of is if the schema file got corrupted. Can you confirm that the schema file `C:\Windows\System32\inetsrv\config\schema\aspnetcore_schema_v2.xml` looks like https://github.com/aspnet/AspNetCore/blob/master/src/Servers/IIS/AspNetCoreModuleV2/AspNetCore/aspnetcore_schema_v2.xml

**Comment by jpseini** [comment]
Thanks again for the suggestion @jkotalik - compared the file in github with the one on my PC and they both exactly the same. Will carry on poking around my side to see if I can find anything.

**Comment by jpseini** [comment]
An update on this for reference @jkotalik  - I installed the exact same app, using the exact same steps on one of our dev servers (as opposed to my local DEV machine), and it seems to be working without issue.

So the issues appears to be environment related. I compared the two "IIS features" installed, and brought my DEV box in sync with the server, but still the same error.

For reference:
Dev machine: Windows 10 Enterprise, IIS 10.0.17763.1
Dev server: Server 2012 R2, IIS 8.5.9600.16384

For now, at least I have it working on the server, but will carry on trying to get it working locally on and off - still not sure if its a "bug" somewhere (related to environment), or if its purely a configuration issue.
